plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -847,6 +847,56 @@ public void testTextualPredicatePushdown()
                 .isFullyPushedDown();
     }
 
+    @Test
+    public void testEnumPredicatePushdown()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase(), "tpch.test_enum", "(evalue Enum('hello', 'world', 'again'), ivalue Int32) ENGINE=Log")) {
+            final String tableName = table.getName();
+
+            assertQuerySucceeds(format("INSERT INTO %s (evalue, ivalue) VALUES(VARCHAR 'hello', 10), (VARCHAR 'world', 20), (VARCHAR 'again', 30)", tableName));
+
+            // Equality check shall be pushed down
+            assertThat(query(format("SELECT ivalue FROM %s WHERE evalue = 'hello'", tableName)))
+                    .matches("VALUES (INTEGER '10')")
+                    .isFullyPushedDown();
+
+            // ClickHouse does support operations like `WHERE evalue < 'world'`, which could
+            // break correctness, therefore inequality and range checks shouldn't be pushed down.
+            // This results in treating enum values as text.
+            assertThat(query(format("SELECT ivalue FROM %s WHERE evalue <> 'hello' ORDER BY ivalue", tableName)))
+                    .matches("VALUES (INTEGER '20'),(INTEGER '30')")
+                    .isNotFullyPushedDown(FilterNode.class);
+
+            assertThat(query(format("SELECT ivalue FROM %s WHERE evalue BETWEEN 'again' AND 'world' ORDER BY ivalue", tableName)))
+                    .matches("VALUES (INTEGER '10'),(INTEGER '20'),(INTEGER '30')")
+                    .isNotFullyPushedDown(FilterNode.class);
+
+            // Enum IN without domain compaction
+            assertThat(query(format("SELECT ivalue FROM %s WHERE evalue IN ('hello', 'again')", tableName)))
+                    .matches("VALUES (INTEGER '10'),(INTEGER '30')")
+                    .isFullyPushedDown();
```

Comment:
Note that `isFullyPushedDown` method verifies the result even without `matches`.